File: minc/value/chunk.go
package value

import "minc/token"

type Chunk struct {
	Code      []byte
	Constants []Value

	Metadata []ChunkMetadata
}

type ChunkMetadata struct {
	Position token.Position
	Length int
}

func NewMetaLen1(pos token.Position) ChunkMetadata {
	return ChunkMetadata{
		Position: pos,
		Length: 1,
	}
}

File: minvm/src/chunk.c
#include "chunk.h"

#include <stdlib.h>
#include <string.h>

void chunk_free(struct chunk *c) {
    free(c->name);
    free(c->code);
    free(c->constants);
    free(c->metadata);
    
    memset(c, 0, sizeof(*c));
}
